### PR TITLE
Unify status badges across Automation pages [MAILPOET-6135]

### PR DESCRIPTION
**Pull request by NeosinneR**
## Description

Unify how different status badges are rendered on Automation pages. This includes:
1. **Automation status** - in Automation editor and on Analytics page
Screenshot 2024-06-26 at 22 02 47
Screenshot 2024-06-26 at 22 02 30

2. **Order status** - on `Analytics > Orders` tab
Screenshot 2024-06-26 at 22 03 46

3. **Subscriber run status** - on `Analytics > Subcribers` tab
Screenshot 2024-06-26 at 22 04 10

4. **Email click status** - on `Analytics > Emails` tab 
Screenshot 2024-06-26 at 22 04 34

---

This PR also adds this badge to the **Automations listing page**.
Screenshot 2024-06-26 at 22 05 09

## Code review notes

_N/A_

## QA notes

Please check that all statuses across Automations pages are rendered properly and color accordingly. Note that `Active` automation is green, but `Complete` Subscriber run or Order are grey. This is on purpose (context in p1718809295291599-slack-C0313447YBC). 

One change in this PR is that the numbers next to the email click badges are black and not in the color of the badge. This was done to simplify the code. 

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6135]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-6135]: https://mailpoet.atlassian.net/browse/MAILPOET-6135?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/automation/components/status/index.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      className={automationRunStatusClasses[status] ?? status}
+    />
+  );
+}
+
+export function OrderStatus({ status, name }: OrderStatusProps): JSX.Element {
+  return (
+    <StatusBadge name={name} className={orderStatusClasses[status] ?? status} />
+  );
+}
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/components/status/index.tsx]: Technically, this one should probably go to the MailPoet integration where it is used (an order is not something known to the automation engine), but it's not a very important thing, we're not always 100% strict with the separation of these. I'll leave the decision up to you.

**Review by JanJakes — DISMISSED** [review]
Nice refactor, thanks!

I left only one comment with a small idea. Feel free to either address it or move this to QA.

File: mailpoet/assets/js/src/automation/components/status/index.tsx (hunk 1 above)
Comment (NeosinneR) [reply mailpoet/assets/js/src/automation/components/status/index.tsx]: @JanJakes you meant something like this? https://github.com/mailpoet/mailpoet/pull/5696/commits/b95ff401fb8db8410e73535ec8ab0cb7c2304a98

**Review by JanJakes — APPROVED** [review]
Great, thanks!

**Review by veljkho — CHANGES_REQUESTED** [review]
Everything's good except the `active` badge in the Emails > Newsletters Stats page [when I click to see the statistics of the automation newsletter](https://d.pr/i/DWVZPI) (so through Automation analytics). I guess this is ok since this `active` is not very much related to the badge in Automation, however, we may consider changing also this badge to unify experience? @NeosinneR

**Comment by NeosinneR** [comment]
@veljkho, a good find. However, I'd keep it as is or consider refactoring it in a separate ticket. I focused on badges inside Automation because it's a separate codebase, and we try not to reuse components from Automations in other MailPoet pages and vice versa. This is because Automations was built with the potential to be extractable from the MailPoet codebase. There are also other badges for clicks and opens on the Emails page that also use different badges, so that would require a bit more effort to unify it everywhere.

**Review by veljkho — APPROVED** [review]
That makes sense. Thanks for clatifying.

**Pull request merged**